### PR TITLE
Python acceleration

### PR DESCRIPTION
**Pull request by pgrinaway**
A brief overview of some of the more common approaches for fast computing in Python. Addresses #7

### CODE REVIEW

**Comment by jchodera** [comment]
Awesome! 

I'd also mention [`pycuda`](https://mathema.tician.de/software/pycuda/) and [`pyopencl`](https://mathema.tician.de/software/pyopencl/) if these are still being maintained.

File: Python_Acceleration.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        replica_states = md.utils.ensure_type(self.replica_states, np.int64, 1, "Replica States")
+        u_kl = md.utils.ensure_type(self.u_kl, np.float64, 2, "Reduced Potentials")
+        Nij_proposed = md.utils.ensure_type(self.Nij_proposed, np.int64, 2, "Nij Proposed")
+        Nij_accepted = md.utils.ensure_type(self.Nij_accepted, np.int64, 2, "Nij accepted")
+        _mix_replicas_cython(self.nstates**4, self.nstates, replica_states, u_kl, Nij_proposed, Nij_accepted)
+```
+
+Note that we have some extra calls to ensure that the type is as we declared. However, it is otherwise as simple as calling a Python function.
+
+## Recommendations
```
Comment (pgrinaway) [inline Python_Acceleration.md]: Yep!

**Comment by pgrinaway** [comment]
>I'd also mention pycuda and pyopencl if these are still being maintained.

Ok, will do

**Comment by pgrinaway** [comment]
Looks like they are still being maintained, so I've added a little section on them too.

**Pull request merged**